tools/js-tools/gen-test-instructions.mjs

```diff
@@ -0,0 +1,574 @@
+		}
+
+		const sectionMatch = line.match( sectionRegex );
+		if ( sectionMatch ) {
+			currentSection = sectionMatch[ 1 ];
+			continue;
+		}
+
+		if ( collectingEntries && currentVersion ) {
+			const entryMatch = line.match( entryRegex );
+			if ( entryMatch ) {
+				entries.push( {
+					text: entryMatch[ 1 ],
+					prNumber: entryMatch[ 2 ],
+					section: currentSection,
+					version: currentVersion,
+					date: currentDate,
+				} );
+			}
+		}
+	}
+
+	return {
+		entries,
+		startVersion: startVersion || 'date: ' + sinceDate,
+		versions: versions.slice( 0, 20 ),
+	};
+}
+
+/**
+ * Extract unique PR numbers from changelog entries.
+ *
+ * @param {Array} entries - Changelog entries
+ * @return {Array} Sorted array of unique PR numbers
+ */
+function extractPRNumbers( entries ) {
+	const prNumbers = new Set();
+	entries.forEach( entry => {
+		if ( entry.prNumber ) {
+			prNumbers.add( entry.prNumber );
+		}
+	} );
+	return Array.from( prNumbers ).sort( ( a, b ) => parseInt( a ) - parseInt( b ) );
+}
+
+// ============================================================================
+// GITHUB PR FETCHING
+// ============================================================================
+
+/**
+ * Fetch PR details from GitHub using gh CLI.
+ *
+ * @param {Array} prNumbers - Array of PR numbers to fetch
+ * @return {Promise<Array>} Array of PR detail objects
+ */
+async function fetchPRDetails( prNumbers ) {
+	const prDetails = [];
+
+	for ( let i = 0; i < prNumbers.length; i++ ) {
+		const prNumber = prNumbers[ i ];
+		process.stdout.write( `\r  Fetching PR #${ prNumber } (${ i + 1 }/${ prNumbers.length })...` );
+
+		try {
+			const prData = execSync(
+				`gh pr view ${ prNumber } --json number,title,body,labels,author --repo ${ GITHUB_REPO }`,
+				{ encoding: 'utf-8', stdio: [ 'pipe', 'pipe', 'ignore' ] }
+			);
+
+			const pr = JSON.parse( prData );
+			const testingInstructions = extractTestingInstructions( pr.body );
+
+			prDetails.push( {
+				number: pr.number,
+				title: pr.title,
+				body: pr.body,
+				testingInstructions,
+				labels: pr.labels.map( l => l.name ),
+				author: pr.author.login,
+			} );
+
+			// Small delay to avoid rate limiting
+			await sleep( 100 );
+		} catch ( error ) {
+			console.warn( `\n⚠️  Could not fetch PR #${ prNumber }: ${ error.message }` );
+		}
+	}
+
+	process.stdout.write( '\r' + ' '.repeat( 80 ) + '\r' ); // Clear the line
+	return prDetails;
+}
+
+/**
+ * Extract testing instructions from PR body.
+ *
+ * @param {string} prBody - PR description
+ * @return {string|null} Testing instructions or null
+ */
+function extractTestingInstructions( prBody ) {
+	if ( ! prBody ) {
+		return null;
+	}
+
+	const patterns = [
+		/## Testing [Ii]nstructions[\s\S]*?(?=\n## |$)/,
+		/### Testing [Ii]nstructions[\s\S]*?(?=\n## |$)/,
+		/## Test [Pp]lan[\s\S]*?(?=\n## |$)/,
+		/### Test [Pp]lan[\s\S]*?(?=\n## |$)/,
+		/## How to [Tt]est[\s\S]*?(?=\n## |$)/,
+		/### How to [Tt]est[\s\S]*?(?=\n## |$)/,
+	];
```

Comment:
Is there a reason we don't want the entire content of the PR? The description often contains lots of valuable context for testing purposes.